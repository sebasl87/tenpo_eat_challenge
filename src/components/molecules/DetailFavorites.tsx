import React from 'react'
import { View } from 'react-native';
import styled from 'styled-components/native';

export interface IDetailFavorites {
    discount?: string;
    restaurantName?: string;
    rate?: string;
    lapseTime?: string;
    favoriteFood?: string;
}

const ImageStar = styled.Image`
    width: 16px;
    height: 16px;
    margin-right: 2px;
`

const DetailFavoritesContainer = styled.View`
    align-items: center;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-bottom: 3px;
    max-width: 260px;
    width: 100%;
`

const RestaurantRate = styled.View`
    flex-direction: row;
    display: flex;
    align-items: center;
`

const ContainerRate = styled.View`
    align-items: center;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    max-width: 260px;
    width: 100%;
`

const NameRestaurant = styled.Text`
    color: #00baa4;
    font-size: 13px;
`

const NameFavoriteFood = styled.Text`
    color: #333333;
    font-size: 13px;
`

const Rate = styled.Text`
    color: #333333;
    font-size: 12px;
`

const LapseTimeText = styled.Text`
    color: #333333;
    font-size: 12px;
`

const GeneralContainer = styled.View`
    width: 100%;
    max-width: 260px;
    display: flex;
    border: 1px solid #EDEDED;
    padding: 8px;
    border-bottom-left-radius: 16px;
    border-bottom-right-radius: 16px;

`;

const DetailFavorites: React.FC<IDetailFavorites> = ({ restaurantName, rate, lapseTime, favoriteFood }) => {
    return (
        <GeneralContainer>
            <DetailFavoritesContainer>
                <NameFavoriteFood style={{ fontFamily: "Gotham-Medium" }}>{favoriteFood}</NameFavoriteFood>
                <RestaurantRate>
                    <ImageStar
                        source={require("../../../assets/imgs/starmini.png")}
                    />
                    <Rate style={{ fontFamily: "Gotham-Medium" }}>{rate}</Rate>
                </RestaurantRate>
            </DetailFavoritesContainer>
            <ContainerRate>
                <NameRestaurant style={{ fontFamily: "Gotham-Bold" }}>{restaurantName}</NameRestaurant>
                <View>
                    <LapseTimeText style={{ fontFamily: "Gotham-Medium" }}>{lapseTime} min</LapseTimeText>
                </View>
            </ContainerRate>
        </GeneralContainer>
    );
}


export default DetailFavorites;