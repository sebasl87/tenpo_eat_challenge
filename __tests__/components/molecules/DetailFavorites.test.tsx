import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import DetailFavorites from "../../../src/components/molecules/DetailFavorites";


describe("DetailFavorites component", () => {
    it("should render successfully with DetailFavorites", () => {
        const titleSection = renderer
            .create(<DetailFavorites favoriteFood="DetailFavorites_mock" rate="DetailFavorites_mock" restaurantName="DetailFavorites_mock" lapseTime="DetailFavorites_mock" />)
            .toJSON();

        expect(titleSection).toMatchSnapshot();
    });
});
