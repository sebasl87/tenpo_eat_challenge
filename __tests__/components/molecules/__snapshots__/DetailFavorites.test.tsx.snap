// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailFavorites component should render successfully with DetailFavorites 1`] = `
<View
  style={
    [
      {
        "borderBottomLeftRadius": 16,
        "borderBottomRightRadius": 16,
        "borderColor": "#EDEDED",
        "borderStyle": "solid",
        "borderWidth": 1,
        "display": "flex",
        "maxWidth": 260,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 3,
          "maxWidth": 260,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#333333",
            "fontSize": 13,
          },
          {
            "fontFamily": "Gotham-Medium",
          },
        ]
      }
    >
      DetailFavorites_mock
    </Text>
    <View
      style={
        [
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/imgs/starmini.png",
          }
        }
        style={
          [
            {
              "height": 16,
              "marginRight": 2,
              "width": 16,
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#333333",
              "fontSize": 12,
            },
            {
              "fontFamily": "Gotham-Medium",
            },
          ]
        }
      >
        DetailFavorites_mock
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "maxWidth": 260,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#00baa4",
            "fontSize": 13,
          },
          {
            "fontFamily": "Gotham-Bold",
          },
        ]
      }
    >
      DetailFavorites_mock
    </Text>
    <View>
      <Text
        style={
          [
            {
              "color": "#333333",
              "fontSize": 12,
            },
            {
              "fontFamily": "Gotham-Medium",
            },
          ]
        }
      >
        DetailFavorites_mock
         min
      </Text>
    </View>
  </View>
</View>
`;
